wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+            # Poor GenCheckBox, Disabled and ignored!
+            if self.IsChecked():
+                return self._bitmaps["CheckedDisable"]
+            else:
+                return self._bitmaps["UnCheckedDisable"]
+
+    def SetLabel(self, label):
+        """
+        Sets the :class:`GenCheckBox` text label and updates the control's
+        size to exactly fit the label plus the bitmap.
+
+        :param `label`: Text to be displayed next to the checkbox.
+        :type `label`: str
+        """
+
+        wx.Control.SetLabel(self, label)
+
+        # The text label has changed, so we must recalculate our best size
+        # and refresh ourselves.
+        self.InvalidateBestSize()
+        self.Refresh()
+
+    def SetFont(self, font):
+        """
+        Sets the :class:`GenCheckBox` text font and updates the control's
+        size to exactly fit the label plus the bitmap.
+
+        :param `font`: Font to be used to render the checkboxs label.
+        :type `font`: `wx.Font`
+        """
+
+        wx.Control.SetFont(self, font)
+
+        # The font for text label has changed, so we must recalculate our best
+        # size and refresh ourselves.
+        self.InvalidateBestSize()
+        self.Refresh()
+
+    def DoGetBestSize(self):
+        """
+        Overridden base class virtual.  Determines the best size of the control
+        based on the label size, the bitmap size and the current font.
+        """
+
+        # Retrieve our properties: the text label, the font and the check
+        # bitmap.
+        label = self.GetLabel()
+        font = self.GetFont()
+        bitmap = self.GetBitmap()
+
+        if not font:
+            # No font defined? So use the default GUI font provided by the system
+            font = self.SYS_DEFAULT_GUI_FONT
+
+        # Set up a wx.ClientDC. When you don't have a dc available (almost
+        # always you don't have it if you are not inside a wx.EVT_PAINT event),
+        # use a wx.ClientDC (or a wx.MemoryDC) to measure text extents.
+        dc = wx.ClientDC(self)
+        dc.SetFont(font)
+
+        # Measure our label.
+        textWidth, textHeight = dc.GetTextExtent(label)
+
+        # Retrieve the check bitmap dimensions.
+        bitmapWidth, bitmapHeight = bitmap.GetWidth(), bitmap.GetHeight()
+
+        # Get the spacing between the check bitmap and the text.
+        spacing = self.GetSpacing()
+
+        # Ok, we're almost done: the total width of the control is simply
+        # the sum of the bitmap width, the spacing and the text width,
+        # while the height is the maximum value between the text width and
+        # the bitmap width.
+        totalWidth = bitmapWidth + spacing + textWidth
+        totalHeight = max(textHeight, bitmapHeight)
+
+        best = wx.Size(totalWidth, totalHeight)
+
+        # Cache the best size so it doesn't need to be calculated again,
+        # at least until some properties of the window change.
+        self.CacheBestSize(best)
```

Comment:
This may be unnecessary. The caller of `DoGetBestSize` should already be passing the size to `CacheBestSize`

Reply:
I'm not sure. by what you said, That basically means to me it is a idiot-proofing line. So you might want to check if what I understand your trying to say is that it may be being called 2twice...? Did not modify this one.